tools/image-info

```diff
@@ -2618,26 +2641,60 @@ def append_partitions(report, image, devpath:str, mountpoint: str):
                 info = index.get_module_info("Mount", "org.osbuild.xfs")
             else:
                 raise RuntimeError("Unknown file system")
-            options = { "readonly":True }
-            for option in part_options:
-                if "=" in option:
-                    parts = option.split("=")
-                    options[parts[0]] = parts[1]
-                else:
-                    options[option] = True
+            if not info:
+                raise RuntimeError(f"Can't find org.osbuild.{part_fstype}")
 
             # the first mount point should be root
             if n == 0:
                 if part_mountpoint != "/":
                     raise RuntimeError("The first mountpoint in sorted fstab entries is not '/'")
                 root_tree = mountpoint
+
+            # prepare the options to mount the partition
+            options = {}
+            for option in part_options:
+                if option == "defaults": # defaults is not a supported option
+                    continue
+
+                if "=" in option:
+                    parts = option.split("=")
+                    key = parts[0]
+                    val = parts[1]
+
+                    # uid and gid must be integers
+                    if key == "uid" or key == "gid":
+                        val = int(val)
+
+                    options[key] = val
+                else:
+                    options[option] = True
+
+            options["readonly"] = True
+
+            # Validate that the options are valid
+            #
+            # The mount manager is taking care of opening the file system for us
+            # so we don't have access to the json objects that'll be used to
+            # invoke the mounter. However we can reconstruct it for the json
+            # validation step. We're only interested at validate the options so
+            # we can put a placeholder string for the other values.
+            validate(info, {
+                "source": "placeholder",
+                "target": "placeholder",
+                "type": "placeholder",
+                "name": "placeholder",
+                "options": options,
+            })
```

Comment:
Would it be simpler to validate `info.get_schema().get("options")` against the `options` since that's the only part we care about?